Scripts/Python/xAgeSDLBoolActivatorComboSet.py

```diff
@@ -52,13 +52,16 @@
 disableOnSolve = ptAttribBoolean(6, "Disable activators on solve", default=True)
 actButtons = ptAttribActivatorList(7, "Act: Buttons")
 respButtonPush = ptAttribResponder(8, "Resp: Button Push")
+allowSlidingSolution = ptAttribBoolean(9, "Allow solving by sliding solution rather than needing feedback on failure", default=False)
 
 class xAgeSDLBoolActivatorComboSet(ptResponder):
     def __init__(self):
         ptResponder.__init__(self)
         self.version = 1
         self.id = 719
 
+        self._attempt = []
```

Comment:
The good thing about this is that it allows Age creators to trivially opt into sliding behaviors without changing anything in their already existing Age using this script. The bad news is that it exposes curious logic in multiplayer. The built in assumption here is that the game master will be present in the Age for every single button press because the window is being cached in Python code rather than persisting on the server. This means that, depending on linking order, two players could perform the same button presses and get different answers for whether or not the puzzle is solved. I think this is arguably worse than the default behavior, which is always consistent in terms of how it accepts and rejects combinations.

I think the best way to solve this would be to change attribute 9 above to an optional SDL variable name in which the combination window is stored. If the window is not specified, fall back to the original behavior. The downside to this is that SDL changes are required, which are always a pain.

Reply:
At risk of opening some sort of can of worms... you can have per-script SDL files rather than per-Age SDL files (see tldnWRCC as an example). Maybe that would be an option here to make it transparent to Age builders?

Reply:
Perhaps - my request on that would be to not remove the age-wide solved and number correct variables. I especially wanted those visible for Age creators to be able to attach things like xAgeSDLBoolShowHide and xAgeSDLIntRespond. 😄

Reply:
Related: the addition of a `STATEDESC xAgeSDLBoolActivatorComboSet` would necessitate the application of the `Blocking on MOULa` tag.

Reply:
I don't plan to touch anything about the existing solved and numCorrect SDL storage. That seems to be working fine. I would like to try the script SDL solution, but I feel a little out of my dept on how to get started with that.

Reply:
I'm currently stuck and need some assistance from @dpogue or @Hoikas to help. I have re-implemented things using the script SDL, and it works as before, but that SDL value isn't appearing to be propagated or shared between multiple users in the Age. Each person interacting with the puzzle gets their own "attemptCombo" SDL that changes independently, which is no better than the original implementation using a local array in Python without involving SDL.

I don't know how to figure out why this isn't getting shared. Maybe this really will need a new Age SDL variable, which will make this whole thing almost entirely unhelpful for existing Ages.